airflow/providers/amazon/aws/utils/waiter_with_logging.py

```diff
@@ -0,0 +1,77 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import logging
+import time
+
+from botocore.exceptions import WaiterError
+
+from airflow.exceptions import AirflowException
+
+
+def wait(
+    waiter,
+    waiter_delay,
+    max_attempts,
+    state_args,
+    failure_message,
+    status_message,
+):
+    log = logging.getLogger(__name__)
+    attempt = 0
+    while attempt < max_attempts:
+        attempt += 1
+        try:
+            waiter.wait(
+                **state_args,
+                WaiterConfig={
+                    "Delay": 1,
```

Comment:
Can we omit this parameter ? If not, maybe add a comment explaining that it's not meaningful here.

Reply:
Yes that parameter can be omitted. As long as `MaxAttempts` is 1, we don't get to the `sleep` call in the waiter.